packages/dart_frog_cli/test/src/route_configuration_watcher/route_configuration_watcher_test.dart

```diff
@@ -0,0 +1,210 @@
+      );
+      expect(routeConfigurationWatcher, isNotNull);
+    });
+  });
+
+  group('start', () {
+    test('starts a watcher successfully.', () async {
+      await expectLater(routeConfigurationWatcher.start(), completes);
+
+      expect(routeConfigurationWatcher.isWatching, isTrue);
+      expect(routeConfigurationWatcher.isRunning, isTrue);
+      expect(routeConfigurationWatcher.isCompleted, isFalse);
+    });
+
+    test('throws when route config watcher is already running', () async {
+      await expectLater(routeConfigurationWatcher.start(), completes);
+
+      await expectLater(
+        routeConfigurationWatcher.start(),
+        throwsA(
+          isA<DartFrogRouteConfigurationWatcherException>().having(
+            (e) => e.toString(),
+            'message',
+            'Cannot start a route config watcher while already running.',
+          ),
+        ),
+      );
+    });
+
+    test('throws when route config watcher has been completed', () async {
+      await expectLater(routeConfigurationWatcher.start(), completes);
+      await routeConfigurationWatcher.stop();
+
+      await expectLater(
+        routeConfigurationWatcher.start(),
+        throwsA(
+          isA<DartFrogRouteConfigurationWatcherException>().having(
+            (e) => e.toString(),
+            'message',
+            'Cannot start a route config watcher after it has been stopped.',
+          ),
+        ),
+      );
+    });
+
+    test('generates a new route config on file changes', () async {
+      final routeConfiguration = _MockRouteConfiguration();
+
+      var onRouteConfigurationChangedCalls = 0;
+
+      routeConfigurationWatcher = RouteConfigurationWatcher(
+        logger: logger,
+        workingDirectory: Directory.current,
+        onRouteConfigurationChanged: (routeConfiguration) {
+          onRouteConfigurationChangedCalls++;
+        },
+        directoryWatcher: (_) => directoryWatcher,
+        routeConfigurationBuilder: (_) => routeConfiguration,
+      );
+
+      await routeConfigurationWatcher.start();
+
+      expect(onRouteConfigurationChangedCalls, equals(0));
+
+      watcherController.add(
+        WatchEvent(
+          ChangeType.MODIFY,
+          path.join(
+            Directory.current.path,
+            'routes',
+            'index.dart',
+          ),
+        ),
+      );
+
+      await Future<void>.delayed(const Duration(milliseconds: 100));
+
+      expect(onRouteConfigurationChangedCalls, equals(1));
+    });
+  });
+
+  group('stop', () {
+    test('stops a watcher successfully', () async {
+      await expectLater(routeConfigurationWatcher.start(), completes);
+      await expectLater(routeConfigurationWatcher.stop(), completes);
+
+      expect(routeConfigurationWatcher.isWatching, isFalse);
+      expect(routeConfigurationWatcher.isRunning, isFalse);
+      expect(routeConfigurationWatcher.isCompleted, isTrue);
+      expect(await routeConfigurationWatcher.exitCode, ExitCode.success);
```

Comment:
```suggestion
      expect(await routeConfigurationWatcher.exitCode, equals(ExitCode.success));
```

Reply:
Does it make sense to use equals on a static value such as a enum?